src/lib/simulation/simulation.ts

```diff
@@ -0,0 +1,73 @@
+import { Alchemy, BigNumber, Network, Utils } from "alchemy-sdk";
+import { EVMMintInstructions } from "../types";
+export const SIGNER1_WALLET = '0x965EF172b303B0BcdC38669DF1De3c26BAD2dB8a';
+export const TEST_RECIPIENT = '0x0cc9601298361e844451a7e35e1d7fcd72750e47';
+
+export const NETWORKS: Record<number, Network> = {
+    1: Network.ETH_MAINNET,
+    5: Network.ETH_GOERLI,
+    137: Network.MATIC_MAINNET,
+    42161: Network.ARB_MAINNET,
+    80001: Network.MATIC_MUMBAI,
+    8453: Network.BASE_MAINNET,
+    84531: Network.BASE_GOERLI,
+    11155111: Network.ETH_SEPOLIA
+};
+
+export const simulateEVMTransactionWithAlchemy = async (mintInstructions: EVMMintInstructions, blockNumber?: string): Promise<{ message: string, success: boolean, rawSimulationResult: any }> => {
+    const alchemy = new Alchemy({
+        apiKey: process.env.ALCHEMY_API_KEY,
+        network: NETWORKS[mintInstructions.chainId],
+    });
+
+    const abi = mintInstructions.abi;
+    const iface = new Utils.Interface(abi);
+
+    const paramsArray = prepareContractParams(mintInstructions.contractParams);
+
+    const data = iface.encodeFunctionData(mintInstructions.contractMethod, paramsArray);
+
+    const tx = {
+        to: mintInstructions.contractAddress,
+        from: SIGNER1_WALLET,
+        data: data,
+        value: BigNumber.from(mintInstructions.priceWei || '0').toHexString().replace('0x0', '0x'),
+    };
+
+    const simResult = await alchemy.transact.simulateExecution(tx, blockNumber ? blockNumber : undefined);
```

Comment:
Alchemy simulations only support a subset of chains, we may want to switch to Tenderly here.

![CleanShot 2024-07-18 at 20 28 46@2x](https://github.com/user-attachments/assets/75c30c40-6d76-49e3-8553-68005d12bbd4)

Reply:
For today, we're mostly only doing Base (Onchain summer bounties) and Ethereum, and have Alchemy credentials in our public repo.

This is also conformant (I believe?) with how we simulate them internally before allowing an in-app purchase to be processed?

All that said, unless there's some reason you believe this is problematic as is, we can move everything to Tenderly as future work.